src/core/Akka/Util/ThreadLocalRandom.cs

```diff
@@ -19,7 +19,9 @@ public static class ThreadLocalRandom
     {
         private static int _seed = Environment.TickCount;
 
-        private static ThreadLocal<Random> _rng = new ThreadLocal<Random>(() => new Random(Interlocked.Increment(ref _seed)));
+        [ThreadStatic] 
+        private static Random _rng2;
+        //private static ThreadLocal<Random> _rng = new ThreadLocal<Random>(() => new Random(Interlocked.Increment(ref _seed)));
```

Comment:
Avoiding ThreadLocal wherever we don't need it for some special reason.

Reasoning here: https://ayende.com/blog/189761-A/production-postmortem-the-slow-slowdown-of-large-systems